### ISSUE TITLE
Firebase push token refreshes after depersonalization

### ISSUE THREAD

**Issue by p1nkydev** [infobip/mobile-messaging-sdk-android#35]
Steps:
1. Personalize (via back-end API)
2. Depersonalize (via SDK)

Expected result: Push token isn't refreshing
Actual result: 3. Push token is refreshed

**Comment by riskpp** [comment]
Hello! @p1nkydev , on the depersonalization we do reset the cloud token, so it's expected behaviour, just to be sure that everything is refreshed after depersonalization.

**Comment by faier-PM** [comment]
Hi everyone,
 
We faced an URGENT issue in production environment.
 
We found out that android SDK initiate push token refresh. It breaks our schema of push notifications for Android. iOS, Huawei work well.
 
Repro Steps:
 
User is trying to log in the app.
App check Firebase push token and update it when it’s needed.
Valid push token stored.
User logged in.
…
User logged out.
InfoBip SDK updates user’s push token in Firebase.
Another Bank system is trying to use last remembered Firebase push token for sending notifications .
The push notification isn’t delivered because of invalid Firebase push token stored on Bank side.
Back to step 1.
 
Regular use case doesn’t require push token refreshing on each login. But now it’s happening on each login. Bank system store new push token when user try to log in.
 
Basically, we can reproduce this issue on our UAT env..
We’ve built 2 app versions and tested them in parallel:
App + InfoBip LiveChat
App without InfoBip LiveChat
 
This issue reproduces only if the android app contains InfoBip SDK. If app doesn’t contain InfoBip SDK, app doesn’t refresh Firebase push token on each login.
 
We’re in process of testing iOS and Huawei builds.

**Comment by riskpp** [comment]
Hi @faier-PM , sorry didn't get, why do you need to call depersonalize on the login? Usually it's called on the logout...
We don’t see that we can do much except of exposing a way of configuring lib not to refresh token on depersonalization, but may be you can refresh token on the "Another Bank system" side or don't call depersonalize on the login?

**Comment by faier-PM** [comment]
Hi @riskpp, we got confused with current logic when this issues reproduces. We'll review it and get back to you in 1 or 2 weeks.

**Comment by tjuric** [comment]
Hi @faier-PM, 
Are there any updates from your side about this issue? 

We'll close it in a week if there'll be no response from your side.

**Comment by faier-PM** [comment]
Hi there,

We’ve already sorted this issue on our side.
Thanks!

чт, 2 лют. 2023 р. о 16:27 Tereza Jurić ***@***.***> пише:

**Comment by tjuric** [comment]
Great news, closing this issue! 
Thanks for the feedback!
